states/segment_anaylsis.go

```diff
@@ -43,26 +44,46 @@ func (app *ApplicationState) StorageAnalysisCommand(ctx context.Context, p *Stor
 	if err != nil {
 		return err
 	}
-
+	output := "storage analysis:\n"
+	totalBinlogCount := 0
+	totalDeltaLogCount := 0
+	totalStatsLogCount := 0
 	for _, segment := range segments {
-		fmt.Printf("segment %d\n", segment.ID)
+		MaybeAppendOutput(&output, fmt.Sprintf("segment %d\n", segment.ID), p)
 		for _, fieldBinlog := range segment.GetBinlogs() {
-			fmt.Println("fieldID:", fieldBinlog.FieldID)
+			MaybeAppendOutput(&output, fmt.Sprintf("fieldID: %d \n", fieldBinlog.FieldID), p)
 			var size int64
 			for _, binlog := range fieldBinlog.Binlogs {
 				info, err := minio.Stat(ctx, binlog.LogPath)
 				if err != nil {
-					fmt.Println("failed to stats", err.Error())
+					MaybeAppendOutput(&output, fmt.Sprintf("failed to stats, %v \n", err.Error()), p)
 					continue
 				}
 				if p.Detail {
-					fmt.Printf("Binlog %s size %s\n", binlog.LogPath, hrSize(info.Size))
+					MaybeAppendOutput(&output, fmt.Sprintf("Binlog %s size %s\n",
+						binlog.LogPath, hrSize(info.Size)), p)
 				}
 				size += info.Size
 			}
-			fmt.Printf("Total binlog size: %s\n", hrSize(size))
+			MaybeAppendOutput(&output, fmt.Sprintf("Total binlog size: %s\n", hrSize(size)), p)
 		}
+		totalBinlogCount += len(segment.GetBinlogs())
+		totalDeltaLogCount += len(segment.GetDeltalogs())
+		totalStatsLogCount += len(segment.GetStatslogs())
```

Comment:
this is the number of FieldBinlog not actually the Binlog number

Reply:
changed